### ISSUE TITLE
Enable artifacts on CI builds

### ISSUE THREAD

**Issue by nsubtil** [KhronosGroup/Vulkan-ValidationLayers#1134]
It would be incredibly useful to be able to pull artifacts from CI builds for VL, especially for sharing drops of the VL binaries with recent improvements for testing.

I'd be particularly interested in the Win32 Release binaries, preferably with PDBs included. I expect others may be interested in Linux binaries as well. A relatively small window for artifact storage would be fine (couple of weeks or so?).

Would it be possible to make artifacts available for CI builds?

**Comment by lunarpapillo** [comment]
Interesting idea.

We currently don't build Windows **RelWithDebInfo** builds, but I think it would be possible with a reasonable effort.

You're thinking of pulling builds from Travis/AppVeyor, right?  Not from some other build source?

**Comment by nsubtil** [comment]
Yes, I'd like to pull builds from AppVeyor myself. I don't think there's any other sources for builds at this point other than AppVeyor/Travis (at least not publicly visible).

**Comment by lunarpapillo** [comment]
AppVeyor does seem to allow storage of artifacts (but it's not clear to me yet whether this service is available on free open-source accounts):

https://www.appveyor.com/docs/packaging-artifacts/

Travis also allows publishing of artifacts, though if I'm reading this correctly, only to an Amazon AWS account:

https://docs.travis-ci.com/user/uploading-artifacts/

If there's community interest in such, we may be able to ask @KhronosGroup to provide resources for such an AWS host.

I'll continue looking into whether AppVeyor's artifact storage is available to open-source accounts.

**Comment by nsubtil** [comment]
I submitted PR #1165 and it works fine with the current AppVeyor setup. It captures the final binaries for each build as artifacts and makes them available (under the small "Artifacts" tab within the build job itself).

That PR does not switch to RelWithDebInfo because I want to understand if there are any relevant changes to build flags other than generating PDBs. But that can be done in parallel once the current PR is merged.

**Comment by lunarpapillo** [comment]
I think with PR #1165 (thanks!) we're set for Windows.  For Linux, we'll need Khronos to invest in AWS storage (at which time we may consider moving Windows storage to AWS as well.)
